astropy/table/tests/test_showtable.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+
```

Comment:
On testing, I don't see tests for `-max-lines, -hide-unit, and -show-dtype`.  Note that max_lines cannot be less than 7, and it applies to the header lines as well.